samples/browser/src/main/java/org/mozilla/samples/browser/MainActivity.kt

```diff
@@ -39,15 +39,15 @@ class MainActivity : AppCompatActivity() {
             components.sessionManager,
             components.sessionUseCases.loadUrl,
             components.defaultSearchUseCase)
-
-        components.sessionIntentProcessor.process(intent)
     }
 
     override fun onResume() {
         super.onResume()
 
         sessionFeature?.start()
         toolbarFeature?.start()
+
+        components.sessionIntentProcessor.process(intent)
```

Comment:
onResume() seems to be wrong. This would be called whenever the app gets to the foreground.

onCreate() (and maybe onNewIntent()?)

Reply:
@pocmo the problem I am having here is that I need the features to start before the intent is being processed. I am not able to make this work with `onCreate` or `onNewIntent`, since we have to start in `onResume`. Any ideas?

Reply:
I assume because you need onCustomTabConfigChanged() to be called in ToolbarPresenter? As mentioned above: I think you want to update the toolbar from initializeView() and not onCustomTabConfigChanged().

Reply:
@pocmo no (I actually missed that comment before). The reason is that we want the ToolbarPresenter to be started, so it's an active Session observer.

Reply:
We need to know (at least currently) when a CustomTab session was added or another session selected when we're processing the intent.

Reply:
Yeah, we need to fix our presenter. The pattern for all those UI components should be: (1) Display current state, (2) update on changes. Looks like we never implemented (1) here. :)

Reply:
OK, this is addressed now. I took a step back and I don't think we should start/stop features in `onPause/onResume` but `onStart/onStop` instead. This is what we do in Fenix. Then this whole type of problem goes away. I think It's reasonable to assume that all features are started when we're processing intents. So, this is cleaner, imo, and actually works :).

Reply:
This may work for now, but it will fall apart for the same reasons again: I think the assumption that all features exist, are started and subscribed at all times is wrong. It's very well possible that we are going to process intents while the toolbar is not even on the screen and we are somewhere in settings. Imagine other components like the tabs tray. There's no need to create, start and subscribe the feature if it's not visible. I think all components will need to display the current state on start and update if there are changes during their lifetime.

Reply:
Apart from that this we would still process the same Intent multiple times in onStart(). Try this:
* Set sample browser as default browser
* Click on a link in another app to open the browser
* Press "home" button of device
* Switch back to app
-> We process the Intent every time the app is shown again. Right now this means we reload the page every time. Once we start opening tabs we will open new tabs every time.

The only way to avoid that is to process the Intent in onCreate() and only if savedInstanceState is null - meaning this activity wasn't just re-created.

Reply:
We can also land this and continue in a follow-up?

Reply:
OK, I will try something. My view on the problem is that the presenters observe and act on changes only after they started. So, they might miss events. However, it should be possible to act on events that are still relevant (that we know occurred and haven't been acted on) when the presenters are started.